### PR TITLE
Added async feature to Linux OS

### PR DESCRIPTION
**Pull request by garydavenport73**
In order to use async play of music in Linux I had to use threads.  There is a main section where a function is that plays the file Linux.  I start that function as a thread.  If block is true, I immediately join the thread so it must complete before its parent continues.  Otherwise it should execute on its own until completion.

### CODE REVIEW

**Comment by garydavenport73** [comment]
I see the project is no longer active.  I plan to continue with this project.  My main goals are to fix the memory leak in windows if it exists and make async work in Linux.  I am pretty sure I know how to do both of these things.  Additionally, I may make a function to stop the sound being played (in async mode only)  Those simple features would allow this program to be used in video games much easier than other programs.  I will likely fork this project to do this, but if someone would like to help with testing the OSX functionality that would be great as I only use Linux and Windows at this point.

**Comment by TaylorSMarks** [comment]
I recommend you look at Boombox - they forked this already and told me basically the same thing.

I agree with you that testing is an enormous issue… if I had a way of running automated tests for this project, I’d be much more willing to accept pull requests on it. When I did support it, I found that most pull requests would break it in one configuration while fixing another (IE, fix special characters in Windows 7 while breaking it for ASCII in Windows 10, or fixing one thing for Python 2.7 while breaking another for Python 3.)

I don’t know how I could easily test all the dozens of different configurations that people want supported…

And so I don’t. Having the list of working platforms remain static seems better than letting it randomly change from version to version.

Taylor

**Comment by garydavenport73** [comment]
Hi Taylor,

Thanks for getting back to me.

Sorry for the late reply.

I think it makes sense to do what you are by sort of freezing the project
for the most part.  Its kind of hard to make it backwards compatible
and forward compatible because so many things change, like modules being
deprecated and so forth.

I do really like your module, so thanks for making it.  When I went from
Windows to Linux I hadn't realized that it only had the sync feature, so
I started looking at the module a little bit to see what I could do.  I am
a software engineering student right now and learned a lot from looking at
your code.

I did my best to address whatever issues I ran into and forked my project
here:

 https://github.com/garydavenport73/playsoundtoo

which contains a single function.

The page does not look much different from yours at the moment, because
when you fork it makes an identical copy.  And without
much thought I named it playsoundtoo, which I plan to change.

Whatever module I make though, I wanted to have a function in it called
playsound with the same arguments.

In other words, I may make a module for example called 'garyssoundplayer'
and then in the code, you can put the statement:
from garyssoundplayer import playsound.  I kind of wanted to keep the
function name the same, so that it would be backwards
compatible with any prior code, except for the import statement.

I really appreciate that idea of making modules without dependencies.

I hope what I have done so far is ok with you, and I can rename all my
functions and github repository to be very different from yours if
you would like.  Of course, when you first fork, the fork is identical, so
I will need to work on making that fork more my own.  But I just
wanted to let you know what is going on.

Thanks again,
Gary
